src/PowerShellEditorServices.Protocol/Server/LanguageServer.cs

```diff
@@ -989,23 +985,59 @@ private static CompletionItemKind MapCompletionKind(CompletionType completionTyp
             BufferRange completionRange)
         {
             string detailString = null;
+            string documentationString = null;
+            string labelString = completionDetails.ListItemText;
 
-            if (completionDetails.CompletionType == CompletionType.Variable)
+            if ((completionDetails.CompletionType == CompletionType.Variable) ||
+                (completionDetails.CompletionType == CompletionType.ParameterName))
             {
-                // Look for variable type encoded in the tooltip
+                // Look for type encoded in the tooltip for parameters and variables.
+                // Display PowerShell type names in [] to be consistent with PowerShell syntax
+                // and now the debugger displays type names.
                 var matches = Regex.Matches(completionDetails.ToolTipText, @"^\[(.+)\]");
+                if ((matches.Count > 0) && (matches[0].Groups.Count > 1))
```

Comment:
This could be simplified by moving the parentheses outside of the square brackets in the regex so that the contents of match group 1 include the brackets.  Not worth making the change if you're already done with the work though :)